Dockerfile

```diff
@@ -60,32 +60,30 @@ RUN dpkg --add-architecture arm64 && \
   apt-get install -y crossbuild-essential-amd64 crossbuild-essential-arm64 git libbtrfs-dev:amd64 libbtrfs-dev:arm64 libseccomp-dev:amd64 libseccomp-dev:arm64
 
 FROM build-base-debian AS build-containerd
+ARG TARGETARCH
 ARG CONTAINERD_VERSION
 RUN git clone https://github.com/containerd/containerd.git /go/src/github.com/containerd/containerd
+RUN TARGETARCH=${TARGETARCH:-amd64}
 WORKDIR /go/src/github.com/containerd/containerd
 RUN git checkout ${CONTAINERD_VERSION} && \
-  mkdir -p /out /out/amd64 /out/arm64 && \
+  mkdir -p /out /out/${TARGETARCH} && \
   cp -a containerd.service /out
 ENV CGO_ENABLED=1
 ENV GO111MODULE=off
 # TODO: how to build containerd as static binaries? https://github.com/containerd/containerd/issues/6158
-RUN GOARCH=amd64 CC=x86_64-linux-gnu-gcc make && \
-  cp -a bin/containerd bin/containerd-shim-runc-v2 bin/ctr /out/amd64
-RUN git clean -xfd
-RUN GOARCH=arm64 CC=aarch64-linux-gnu-gcc make && \
-  cp -a bin/containerd bin/containerd-shim-runc-v2 bin/ctr /out/arm64
+RUN GOARCH=${TARGETARCH} CC=`[ "${TARGETARCH}" = "arm64" ] && echo aarch64-linux-gnu-gcc || echo x86_64-linux-gnu-gcc` make && \
+  cp -a bin/containerd bin/containerd-shim-runc-v2 bin/ctr /out/${TARGETARCH}
 
 FROM build-base-debian AS build-runc
 ARG RUNC_VERSION
+ARG TARGETARCH
 RUN git clone https://github.com/opencontainers/runc.git /go/src/github.com/opencontainers/runc
 WORKDIR /go/src/github.com/opencontainers/runc
 RUN git checkout ${RUNC_VERSION} && \
   mkdir -p /out
 ENV CGO_ENABLED=1
-RUN GOARCH=amd64 CC=x86_64-linux-gnu-gcc make static && \
-  cp -a runc /out/runc.amd64
-RUN GOARCH=arm64 CC=aarch64-linux-gnu-gcc make static && \
-  cp -a runc /out/runc.arm64
+RUN TARGETARCH=${TARGETARCH:-amd64} GOARCH=${TARGETARCH} CC=`[ "${TARGETARCH}" = "arm64" ] && echo aarch64-linux-gnu-gcc || echo x86_64-linux-gnu-gcc` make static && \
+  cp -a runc /out/runc.${TARGETARCH}
```

Comment:
Shouldn't we use https://github.com/tonistiigi/xx

Reply:
I had no idea about this; let me check this out.
Thanks! Will get back.